core/src/main/java/org/testcontainers/dockerclient/transport/TestcontainersDockerCmdExecFactory.java

```diff
@@ -0,0 +1,304 @@
+package org.testcontainers.dockerclient.transport;
```

Comment:
check the javadoc for this class. This is almost one-to-one version from docker-java with some modifications.
The whole class to be replaced with another transport (like OkHttp), so I wouldn't bother refactoring it any futher

Reply:
👍 I'll not review this in too much detail as it's largely based on well-tested principles.